src/lib/flows/share/steps/share-url.svelte

```diff
@@ -0,0 +1,213 @@
+      icon: XIcon,
+      href: `https://twitter.com/intent/tweet?text=${encodeURIComponent(text)}&url=${encodeURIComponent(url)}`,
+    },
+    {
+      name: 'Facebook',
+      icon: FacebookIcon,
+      href: `http://www.facebook.com/sharer/sharer.php?u=${encodeURIComponent(url)}`,
+    },
+    {
+      name: 'Threads',
+      icon: ThreadsIcon,
+      href: `https://threads.net/intent/post?text=${encodeURIComponent(text)}${encodeURIComponent(url)}`,
+    },
+    {
+      name: 'Telegram',
+      icon: TelegramIcon,
+      href: `https://t.me/share/url?url=${encodeURIComponent(url)}&text=${encodeURIComponent(text)}`,
+    },
+    {
+      name: 'WhatsApp',
+      icon: WhatsAppIcon,
+      href: `https://wa.me/?text=${encodeURIComponent(text)}${encodeURIComponent(url)}`,
+    },
+    {
+      name: 'Email',
+      icon: EmailIcon,
+      href: `mailto:?body=${encodeURIComponent(url)}`,
+    },
+    {
+      name: 'More',
+      icon: EllipsisIcon,
+      onClick() {
+        if (shareSupported) {
+          navigator.share({
+            text,
+            url,
+          });
+        } else {
+          copyShareLink();
+        }
+      },
+    },
+  ];
+
+  function downloadImage() {
+    const segments = new URL(window.location.origin + downloadableImageUrl + text).pathname.split(
+      '/',
+    );
+    const last = segments.pop() || segments.pop(); // Handle potential trailing slash
+    downloadUrl(downloadableImageUrl, last);
+  }
+</script>
+
+<StepLayout>
+  <div class="share-url">
+    {#if downloadableImageUrl}
+      <div class="downloadable-image">
+        <div class="downloadable-image__card">
+          <img src={downloadableImageUrl} alt="downloadable header" />
+        </div>
+        <Button on:click={downloadImage} variant="normal" icon={DownloadIcon}>Download</Button>
+      </div>
+    {/if}
+    <div class="share-options pixelated">
+      <h2 class="pixelated">Share</h2>
+      <p>{shareModalText}</p>
+      <div class="share-options__options">
+        <Button
+          on:mouseenter={() => (hovering = true)}
+          on:focus={() => (hovering = true)}
+          on:mouseleave={() => (hovering = false)}
+          on:blur={() => (hovering = false)}
+          on:click={copyShareLink}
+        >
+          <div class="button-inner">
+            <div class="icon">
+              {#if copySuccess}
+                <span transition:fade={{ duration: 200 }}>
+                  <CheckCircleIcon style="fill: var(--color-positive)" />
+                </span>
+              {:else if hovering}
+                <span transition:fade={{ duration: 200 }}>
+                  <CopyIcon style="fill: currentColor" />
+                </span>
+              {:else}
+                <span transition:fade={{ duration: 200 }}
+                  ><LinkIcon style="fill: currentColor" /></span
+                >
+              {/if}
+            </div>
+            Copy link
+          </div>
+        </Button>
```

Comment:
i think would be slightly nicer if the copy link button was a separate `copy-link-button.svelte` component which is only imported here, just so that the `hovering` state which is only for the copy linking button is more neatly encapsulated and clearly only for the copy link button

Reply:
Makes, total sense will do!